R/ParamHandle.R

```diff
@@ -6,14 +6,144 @@
 #'
 #' @return [\code{\link{ParamHandle}}].
 #' @family ParamHelpers
+#' @export
+
 ParamHandle = R6Class("ParamHandle",
   inherit = ParamBase, # FIXME: Are we sure? Yes!
   public = list(
-
-    root = NULL,
+    # member variables
+    id = NULL,
+    node = NULL, # simple ParamNode
+    val = NULL,  # node value
+    flatval = NULL, # if the node is itself a tree, this hold the preroot traversal of the tree
+    depend = NULL,  # gamma param is valid only when kernel = "RBF"
+    require.expr = NULL,
+    #
+    parent = NULL,
+    root = NULL,    # root has to be changed when parent changed!
+    reldepth = 0L,  # reldepth has to be updated when parent changed!
+    #
+    mand.children = NULL,
+    cond.children = NULL,
+    visitor = NULL,
```

Comment:
What does `visitor` mean?

Reply:
it is a class that visit this handle, google "Visitor design pattern"

Reply:
Okay. I begin to understand.